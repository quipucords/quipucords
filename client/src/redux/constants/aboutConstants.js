const ABOUT_DIALOG_OPEN = 'ABOUT_DIALOG_OPEN';
const ABOUT_DIALOG_CLOSE = 'ABOUT_DIALOG_CLOSE';

export { ABOUT_DIALOG_OPEN, ABOUT_DIALOG_CLOSE };
