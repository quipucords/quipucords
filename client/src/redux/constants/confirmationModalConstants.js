const CONFIRMATION_MODAL_SHOW = 'CONFIRMATION_MODAL_SHOW';
const CONFIRMATION_MODAL_HIDE = 'CONFIRMATION_MODAL_HIDE';

export { CONFIRMATION_MODAL_SHOW, CONFIRMATION_MODAL_HIDE };
