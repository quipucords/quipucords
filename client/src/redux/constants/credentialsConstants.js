
const LOAD_CREDENTIALS_SUCCESS = 'LOAD_CREDENTIALS_SUCCESS';

export {
  LOAD_CREDENTIALS_SUCCESS
}
