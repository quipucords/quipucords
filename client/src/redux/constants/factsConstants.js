const ADD_FACTS_ERROR = 'ADD_FACTS_ERROR';
const ADD_FACTS_LOADING = 'ADD_FACTS_LOADING';
const ADD_FACTS_SUCCESS = 'ADD_FACTS_SUCCESS';

export { ADD_FACTS_ERROR, ADD_FACTS_LOADING, ADD_FACTS_SUCCESS };
