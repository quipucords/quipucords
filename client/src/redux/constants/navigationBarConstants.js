const NAV_TOGGLE_COLLAPSE = 'NAV_TOGGLE_COLLAPSE';
const NAV_COLLAPSE = 'NAV_COLLAPSE';
const NAV_OPEN = 'NAV_OPEN';

export { NAV_TOGGLE_COLLAPSE, NAV_COLLAPSE, NAV_OPEN };
