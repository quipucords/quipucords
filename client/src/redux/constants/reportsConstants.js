const GET_REPORT = 'GET_REPORT';
const GET_MERGE_REPORT = 'GET_MERGE_REPORT';

export { GET_REPORT, GET_MERGE_REPORT };
