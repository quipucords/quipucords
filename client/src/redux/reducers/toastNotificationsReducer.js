import { toastNotificationTypes } from '../constants';

const initialState = {
  toasts: [],
  paused: false
};

const toastNotificationsReducer = (state = initialState, action) => {
  switch (action.type) {
    case toastNotificationTypes.TOAST_ADD:
      const newToast = {
        header: action.header,
        message: action.message,
        alertType: action.alertType,
        persistent: action.persistent
      };

      return Object.assign({}, state, {
        toasts: [...state.toasts, newToast],
        displayedToasts: state.displayedToasts + 1
      });

    case toastNotificationTypes.TOAST_REMOVE:
      const index = state.toasts.indexOf(action.toast);
      action.toast.removed = true;

      const displayedToast = state.toasts.find(toast => !toast.removed);

      if (!displayedToast) {
        return Object.assign({}, state, { toasts: [] });
      }

      return Object.assign({}, state, {
        toasts: [...state.toasts.slice(0, index), action.toast, ...state.toasts.slice(index + 1)]
      });

    case toastNotificationTypes.TOAST_PAUSE:
      return Object.assign({}, state, { paused: true });

    case toastNotificationTypes.TOAST_RESUME:
      return Object.assign({}, state, { paused: false });

    default:
      return state;
  }
};

toastNotificationsReducer.initialState = initialState;

export { toastNotificationsReducer as default, initialState, toastNotificationsReducer };
