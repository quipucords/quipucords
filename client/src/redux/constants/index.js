import * as aboutTypes from './aboutConstants';
import * as credentialsTypes from './credentialsConstants';
import * as factsTypes from './factsConstants';
import * as navigationBarTypes from './navigationBarConstants';
import * as reportsTypes from './reportsConstants';
import * as scansTypes from './scansConstants';
import * as sourcesTypes from './sourcesConstants';
import * as viewToolbarTypes from './viewToolbarConstants';

const reduxTypes = {
  about: aboutTypes,
  credentials: credentialsTypes,
  facts: factsTypes,
  navigation: navigationBarTypes,
  reports: reportsTypes,
  scans: scansTypes,
  sources: sourcesTypes,
  viewToolbar: viewToolbarTypes
};

export {
  reduxTypes,
  aboutTypes,
  credentialsTypes,
  factsTypes,
  navigationBarTypes,
  reportsTypes,
  scansTypes,
  sourcesTypes,
  viewToolbarTypes
};

export default reduxTypes;
