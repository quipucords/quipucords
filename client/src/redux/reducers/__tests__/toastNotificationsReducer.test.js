import { toastNotificationsReducer } from '../toastNotificationsReducer';
import { toastNotificationTypes as types } from '../../constants';

describe('toastNotificationsReducer', () => {
  it('should return the initial state', () => {
    expect(toastNotificationsReducer.initialState).toBeDefined();
  });

  it('should handle specific defined types', () => {
    const specificTypes = [types.TOAST_ADD, types.TOAST_REMOVE, types.TOAST_PAUSE, types.TOAST_RESUME];

    specificTypes.forEach(value => {
      const dispatched = {
        type: value
      };

      const resultState = toastNotificationsReducer(undefined, dispatched);

      expect({ type: value, result: resultState }).toMatchSnapshot(`defined type ${value}`);
    });
  });

  it('should handle adding and removing toast notifications', () => {
    let dispatched = {
      type: types.TOAST_ADD,
      header: 'Lorem',
      message: 'Lorem ipsum dolor',
      alertType: 'success'
    };

    let resultState = toastNotificationsReducer(undefined, dispatched);
    resultState = toastNotificationsReducer(resultState, dispatched);

    expect(resultState).toMatchSnapshot('toast added');

    dispatched = {
      type: types.TOAST_REMOVE,
      toast: resultState.toasts[0]
    };

    resultState = toastNotificationsReducer(resultState, dispatched);
    expect(resultState).toMatchSnapshot('toast removed');
  });
});
