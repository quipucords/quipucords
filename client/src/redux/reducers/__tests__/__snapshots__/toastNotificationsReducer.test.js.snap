// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defined type TOAST_ADD 1`] = `
Object {
  "result": Object {
    "displayedToasts": 1,
    "paused": false,
    "toasts": Array [
      Object {
        "alertType": undefined,
        "header": undefined,
        "message": undefined,
        "persistent": undefined,
      },
    ],
  },
  "type": "TOAST_ADD",
}
`;

exports[`defined type TOAST_PAUSE 1`] = `
Object {
  "result": Object {
    "displayedToasts": 0,
    "paused": true,
    "toasts": Array [],
  },
  "type": "TOAST_PAUSE",
}
`;

exports[`defined type TOAST_REMOVE 1`] = `
Object {
  "result": Object {
    "displayedToasts": 0,
    "paused": false,
    "toasts": Array [],
  },
  "type": "TOAST_REMOVE",
}
`;

exports[`defined type TOAST_RESUME 1`] = `
Object {
  "result": Object {
    "displayedToasts": 0,
    "paused": false,
    "toasts": Array [],
  },
  "type": "TOAST_RESUME",
}
`;

exports[`toast added 1`] = `
Object {
  "displayedToasts": 2,
  "paused": false,
  "toasts": Array [
    Object {
      "alertType": "success",
      "header": "Lorem",
      "message": "Lorem ipsum dolor",
      "persistent": undefined,
    },
    Object {
      "alertType": "success",
      "header": "Lorem",
      "message": "Lorem ipsum dolor",
      "persistent": undefined,
    },
  ],
}
`;

exports[`toast removed 1`] = `
Object {
  "displayedToasts": 2,
  "paused": false,
  "toasts": Array [
    Object {
      "alertType": "success",
      "header": "Lorem",
      "message": "Lorem ipsum dolor",
      "persistent": undefined,
      "removed": true,
    },
    Object {
      "alertType": "success",
      "header": "Lorem",
      "message": "Lorem ipsum dolor",
      "persistent": undefined,
    },
  ],
}
`;
